src/notifications.ts

```diff
@@ -28,25 +66,45 @@ export interface ViolatedLimit {
   current_value: number
 }
 
-export interface ProjectExceedingTierLimit {
+export type ProjectExceedingTierLimitData = {
+  name: NotificationName.ProjectExceedingTierLimit
   violations: ViolatedLimit[]
 }
 
-export interface PostgresRestartRequired {
-  reason: string
-  forceRestartBy?: Date
+// PG Upgrades
+
+export interface ExtensionsUpgrade {
+  name: string
+  version_to: string
 }
 
-export interface SharedMeta {
-  actions: Actions[]
+export interface ServerUpgrade {
+  version_to: string
 }
 
-export interface Notification {
-  id: string
-  project_id: number
-  inserted_at: Date
-  notification_name: NotificationName
-  notification_status: NotificationStatus
-  data: any
-  meta: any
+// TODO (darora): config upgrades would also be included here, but we
+// need to create a config versioning and tracking scheme first
+
+export type PostgresqlUpgradeAvailableData = {
+  upgrade_type: 'postgresql-server' | 'extensions'
+  additional: ExtensionsUpgrade | ServerUpgrade
+}
+
+// PG Upgrade completed
+
+export type PostgresqlUpgradeCompletedData = {
```

Comment:
compress `PostgresqlUpgradeCompletedData` and `PostgresqlUpgradeAvailableData` to just `PostgresqlUpgradeData`?

unless we foresee "completed" and "available" states will have different contents

Reply:
thinking additional can also just be one interface?
```
{ name?: string, version_to: string }
```

Reply:
I've separated them since imo we shouldn't be switching a "seen" notification to "new"; in this case we would want the completed notification to be new?

Reply:
wrt additional - config is very likely to have additional, different fields, and I'd rather not get in the game of 80% of the fields being optional based on which entity got upgraded

Reply:
gotcha on this - makes sense!